bridges/snowbridge/primitives/router/src/outbound/mod.rs

```diff
@@ -71,6 +71,8 @@ impl<UniversalLocation, EthereumNetwork, OutboundQueue, AgentHashedDescription,
 		let dest = destination.take().ok_or(SendError::MissingArgument)?;
 		if dest != Here {
 			log::trace!(target: "xcm::ethereum_blob_exporter", "skipped due to unmatched remote destination {dest:?}.");
+			// We need to make sure that destination is not consumed in case of `NotApplicable`.
+			*destination = Some(dest);
```

Comment:
@claravanstaden 
bellow there is a another usage:
```
if Ok(local_net) != universal_location.global_consensus() {
			log::trace!(target: "xcm::ethereum_blob_exporter", "skipped due to unmatched relay network {local_net:?}.");
			return Err(SendError::NotApplicable)
		}
```

Reply:
I wondered about that one, because the [comment](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-executor/src/traits/export.rs#L44) only mentions dest and msg. But basically no mutable parameters should be altered when the error is `NotApplicable`?

Reply:
Fixed in https://github.com/paritytech/polkadot-sdk/pull/5789/commits/bfc90ba5e9b8f55bf9e13210d06fe66f0e7b50dd. This case is easy to miss... Perhaps the parameters should not be mutable in the first place?

Reply:
I replaced it with a clone, as @yrong suggested. :) https://github.com/paritytech/polkadot-sdk/pull/5789/commits/4fe408539e2ed56c4e10168059dc2318ac7abeb1